pyseir/models/seir_model.py

```diff
@@ -263,45 +261,48 @@ def _time_step(self, y, t):
+               - infected_and_in_hospital_icu
 
-        recovered_after_hospital_general = HNonICU / self.hospitalization_length_of_stay_general
-        recovered_after_hospital_icu = HICU * ((1 - self.fraction_icu_requiring_ventilator)/ self.hospitalization_length_of_stay_icu
-                                               + self.fraction_icu_requiring_ventilator / self.hospitalization_length_of_stay_icu_and_ventilator)
+        mortality_rate_ICU = self.mortality_rate_from_ICU if HICU <= self.beds_ICU else self.mortality_rate_no_ICU_beds
+        mortality_rate_NonICU = self.mortality_rate_from_hospital if HNonICU <= self.beds_general else self.mortality_rate_no_general_beds
 
-        dHNonICU_dt = infected_and_in_hospital_general - recovered_after_hospital_general
-        dHICU_dt = infected_and_in_hospital_icu - recovered_after_hospital_icu
+        died_from_hosp = HNonICU * mortality_rate_NonICU / self.hospitalization_length_of_stay_general
+        died_from_icu = HICU * (1 - self.fraction_icu_requiring_ventilator) * mortality_rate_ICU / self.hospitalization_length_of_stay_icu
+        died_from_icu_vent = HICUVent * self.mortality_rate_from_ICUVent / self.hospitalization_length_of_stay_icu_and_ventilator
 
-        # Tracking categories...
-        dTotalInfections = exposed_and_symptomatic + exposed_and_asymptomatic
-        dHAdmissions_general = infected_and_in_hospital_general
-        dHAdmissions_ICU = infected_and_in_hospital_icu  # Ventilators also count as ICU beds.
+        recovered_after_hospital_general = HNonICU * (1 - mortality_rate_NonICU) / self.hospitalization_length_of_stay_general
 
+        recovered_from_icu_no_vent = HICU * (1 - mortality_rate_ICU) * (1 - self.fraction_icu_requiring_ventilator) / self.hospitalization_length_of_stay_icu
+        recovered_from_icu_vent = HICU * (1 - max(mortality_rate_ICU, self.mortality_rate_from_ICUVent)) \
+                                  * self.fraction_icu_requiring_ventilator / self.hospitalization_length_of_stay_icu_and_ventilator
 
-        # This compartment is for tracking ventillator count. The beds are accounted for in the ICU cases.
+        dHNonICU_dt = infected_and_in_hospital_general - recovered_after_hospital_general - died_from_hosp
+        dHICU_dt = infected_and_in_hospital_icu - recovered_from_icu_no_vent - recovered_from_icu_vent - died_from_icu - died_from_icu_vent
+
+        # This compartment is for tracking ventillator count. The beds are
+        # accounted for in the ICU cases.
         dHICUVent_dt = infected_and_in_hospital_icu * self.fraction_icu_requiring_ventilator \
                        - HICUVent / self.hospitalization_length_of_stay_icu_and_ventilator
 
+        # Tracking categories...
+        dTotalInfections = exposed_and_symptomatic + exposed_and_asymptomatic
+        dHAdmissions_general = infected_and_in_hospital_general
+        dHAdmissions_ICU = infected_and_in_hospital_icu  # Ventilators also count as ICU beds.
```

Comment:
What is the purpose of this? This number will keep increasing since it does not track those who recovered. Will this be helpful to inform intervention decisions?

Reply:
DJ wanted to know how many ICU admissions per day there would be at one point.  And we differentiate this to get the per-day piece.